### PR TITLE
updated scikit-learn to 0.24.2

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
Hey @paxcema, I am trying to implement the second solution that you have mentioned here,
https://github.com/mindsdb/mindsdb/pull/3675

I installed the newer version of `scikit-learn` and ran the tests available in the repository. They seem to be able to run without a problem,
image

image

What do you think? Do you have any concerns?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ python = ">=3.7,<3.10"
 type-infer = "^0.0.9"
 numpy = "^1"
 pandas = "^1"
-scikit-learn = ">=1.0.0, <=1.0.2"
+scikit-learn = "0.24.2"
```
Comment (paxcema) [inline pyproject.toml]: Let's change this to `>= 0.24.2`, that way you can finish the handlers that require this while enabling other users to keep the newest (and better) versions of sklearn.

**Review by paxcema — CHANGES_REQUESTED** [review]
Hey @MinuraPunchihewa!

Awesome to see those tests passing. I've left a single request, but after that we could merge this change.

File: pyproject.toml (hunk 1 above)
Comment (MinuraPunchihewa) [reply pyproject.toml]: Hey @paxcema, you are absolutely right.
I just wanted to know, will changing only the `pyproject.toml` file solve this? Or will we need to make changes to the `poetry.lock` as well?
I think we might need to execute `poetry add  "scikit-learn>=0.24.2"` and allow the relevant changes to be made to the files?

Comment (paxcema) [reply pyproject.toml]: Indeed, we should update the `poetry.lock` file. I believe you can manually change the `.toml`, then do `poetry lock` from the root folder in your terminal to update the file (you could also remove it and generate it from scratch, not sure which approach is faster but it shouldn't take too long).

Comment (MinuraPunchihewa) [reply pyproject.toml]: @paxcema I just pushed another commit. Please take a look.

**Review by paxcema — APPROVED** [review]

**Comment by MinuraPunchihewa** [comment]
Hey @paxcema, thank you for approving this. Does the package automatically get released to PyPI? Or will I have to install it from the source code?

**Comment by paxcema** [comment]
Thank you @MinuraPunchihewa for the contribution! I will be releasing this later in the day to PyPI. In the meantime I would recommend you install from source if you need to continue working on PyCaret.

**Comment by MinuraPunchihewa** [comment]
@paxcema Got it. Thank you very much!

**Pull request merged**